kubernetes/schema_horizontal_pod_autoscaler.go

```diff
@@ -194,7 +225,7 @@ func metricSpecFields() *schema.Resource {
 			"type": {
 				Type:        schema.TypeString,
 				Required:    true,
-				Description: `type is the type of metric source. It should be one of "Object", "Pods", "External" or "Resource", each mapping to a matching field in the object.`,
+				Description: `type is the type of metric source. It should be one of "ContainerResource", "External", "Object", "Pods" or "Resource", each mapping to a matching field in the object. Note: "ContainerResource" type is available on when the feature-gate HPAContainerMetrics is enabled`,
```

Comment:
Let's add validation to this attribute to check that it's one of the values documented here. You can use [validation.StringInSlice](https://pkg.go.dev/github.com/hashicorp/terraform/helper/validation#StringInSlice)